api/api/admin/__init__.py

```diff
@@ -127,3 +138,32 @@ class AccessTokenAdmin(admin.ModelAdmin):
         "created",
         "updated",
     )
+
+
+@admin.register(UserPreferences)
+class IndividualUserPreferencesAdmin(admin.ModelAdmin):
+    """
+    Model admin for showing user preferences. This should only ever show the
+    currently logged-in user's preferences
+    """
+
+    can_delete = False
+    verbose_name_plural = "My Preferences"
+    verbose_name = "My Preferences"
+    form = UserPreferencesAdminForm
+
+    def get_queryset(self, request):
+        qs = super().get_queryset(request)
+        return qs.filter(user=request.user)
+
+    def has_change_permission(self, request, obj=None):
+        if not obj:
+            # the changelist itself
+            return True
+        return obj.user == request.user
```

Comment:
This is not needed, as any objects that are not associated with the current user have already been removed from the queryset and automatically do not have the change permission. Additionally we need to remove permissions to add and delete instances.

```suggestion
    def has_add_permission(self, request):
        return False

    def has_delete_permission(self, request, obj=None):
        return False
```

Reply:
It looks like this is required @dhruvkb, see 1f64f2888be569ecb4650ab18269a699677c4e53.